client/src/pages/course/mentor/confirm.tsx

```diff
@@ -75,21 +78,65 @@ function Page(props: { session: Session; courseAlias?: string }) {
     }
   };
 
-  if (course == null || (mentorData == null && loading)) {
+  if (loading) {
     return null;
   }
 
+  if (course == null || (mentorData == null && loading)) {
+    return (
+      <PageLayout loading={loading} githubId={props.session.githubId}>
+        <Row justify="center" style={{ margin: '65px 0 25px 0' }}>
+          <Image src="/static/svg/err.svg" alt="Error 404" width={175} height={175} />
+        </Row>
+        <Row justify="center">
+          <h1 style={{ fontSize: '36px', marginBottom: 0 }}>Sorry, Course Not Found</h1>
+        </Row>
+      </PageLayout>
+    );
+  }
+
   const pageProps = {
     loading,
     title: 'Confirm Mentorship',
     githubId: props.session.githubId,
     courseName: course.name,
   };
 
-  if (noAccess) {
+  if (noAccess && !isPreferredCourse) {
     return (
       <PageLayout {...pageProps}>
-        <Result status={'403' as any} title="You are not preselected to the course" />
+        <Result status={'403' as any} title="You are not registered to the course" />
+      </PageLayout>
+    );
+  }
+
+  if (noAccess && isPreferredCourse) {
+    return (
+      <PageLayout loading={false} githubId={props.session.githubId}>
+        <Row justify="center" style={{ margin: '65px 0 25px 0' }}>
+          <Image src="/static/images/rs-hero.png" alt="Error 404" width={175} height={175} />
```

Comment:
Why alt text is `Error 404`? Maybe there should be something like `You are RS hero!`

Reply:
fixed